darts/preprocessing/base_transformer.py

```diff
@@ -0,0 +1,238 @@
+"""
+Base Transformer
+----------------
+"""
+from abc import ABC, abstractmethod
+from typing import TypeVar, Generic, Callable, Optional, List, Tuple
+
+from darts.logging import raise_if_not, get_logger
+from darts.preprocessing.validator import Validator
+
+logger = get_logger(__name__)
+T = TypeVar('T')
+
+
+class BaseTransformer(Generic[T], ABC):
+    def __init__(self,
+                 validators: Optional[List[Validator]] = None,
+                 validator_fns: Optional[List[Tuple[Callable[[T], bool], Optional[str]]]] = None,
+                 reversible: bool = False,
+                 can_predict: bool = False,
+                 fittable: bool = False):
+        """
+        Abstract class for transformers. All deriving classes have to implement only one function `transform`.
+        It also have `inverse_transform`, `fit` and `predict` left unimplemented. If child of this class implements
+        any of this methods it should mark this with appropriate property (`reversible`, `fittable`, `can_predict`).
```

Comment:
```suggestion
        any of these methods, it should mark them with the appropriate property (`reversible`, `fittable`, `can_predict`).
```
(sorry for being annoying)

Reply:
Thanks for pointing this and all the rest mistakes/typos. Don't be sorry these are all valuable comments to me, thanks again